ratelimit/redis.go

```diff
@@ -0,0 +1,166 @@
+package ratelimit
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/go-redis/redis"
+	log "github.com/sirupsen/logrus"
+)
+
+// RedisOptions is used to configure the redis.Ring
+type RedisOptions struct {
+	// Addrs are the list of redis shards
+	Addrs []string
+}
+
+// clusterLimitRedis stores all data required for the cluster ratelimit.
+type clusterLimitRedis struct {
+	group   string
+	maxHits int64
+	window  time.Duration
+	ring    *redis.Ring
```

Comment:
Didn't all rings have the same options? I'm would not keep a single `redis.Ring` in the Registry unless there is a difference.

Reply:
because it's already mentioned at the top of ratelimits (missing a -, but it shows `enable-ratelimits`).

Reply:
Yes, but I did it per ratelimit instance. So your suggestion is to have it in registry and pass the isntance to the ratelimiter on creation?

Reply:
The idea is to reduce the memory footprint here. However, I'm not sure how much overhead this creates.

Reply:
I changed it in the latest commit